NewForce-Cohort-9/tabloidmvc-stafford-infection-1
# Delete comment

# Description

Added functionality Delete Comment that allows you to delete a comment for a specific post. This allows you to delete a Comment from a Post that you have created so that you can prevent others from seeing it in the event that you regret what you said.

Trello Ticket: https://trello.com/c/MrQRhA73/21-21-delete-a-comment
Type of change: New feature (non-breaking change which adds functionality)


## Type of change

Please delete options that are not relevant.


# Testing Instructions

1. Ensuring that you have cloned down the repo and added the database, git fetch --all in the master branch to retrieve my branch named fw-create-comment-ticket21
2. git checkout fw-create-comment-ticket21 to switch to my branch. If you don't see the most recent version of my files, git pull origin fw-create-comment-ticket21
3. Open up Visual Studio. 
4. Run the application in a browser window
5. You should see the 'Home' page with a navigation bar at the top of the page. Select 'Login' and log in with this email: [admin@example.com](mailto:admin@example.com)
6. Go to Posts on the nav bar
7. You will see an “eye” icon/button, click on it to go to Post Details 
8. Given you are viewing the Details of a Post, when you select the View Comments button (two way conversation messaging bubbles icon) 
9. You are directed to the Comments list page for the Post
10. Each Comment displays the following information: Subject, Content, Author's Display Name, Creation date, as well as edit and delete buttons. 
11. For testing purposes, you can delete the comment you just created in the previous step for the Create Comment pull request, that is the comment with Subject: “Test”, Content:“Testing PR”.
12. Click on the Delete option for that comment.
13. You are directed to another page which will prompt you, “Are you sure you want to delete this?” in order to confirm your decision to delete the selected comment. 
14. You will see details of the comment including Subject, Content, Author's Display Name and Creation date to review that you are deleting the comment that you intended.
15. If you change your mind, there is a link  “[Cancel Delete - Back to List](https://localhost:7131/Comment?postId=1)” that rejects the confirmation so that the comment is NOT removed from the system and you will be directed back to the Comment list.
16. If you are firm with your decision to delete the comment, and for testing purposes, click on the red Delete button to delete the comment and the Comment will be removed from the system. You will then be directed back to the Comment list.
17. Ensure that the comment with Subject: “Test”, Content:“Testing PR” has been deleted in the Comment list.

# Checklist:
